### PR TITLE
[FLINK-26390] Use try resourses for CliClient#executeFile

### PR DESCRIPTION
**Pull request by snuyanzin**
## What is the purpose of the change

The PR makes use of try resources for `CliClient#executeFile` and skip creation of `Completer` object in case of `CliClient#executeFile` since `Completer` makes sense only in interactive mode

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): ( no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: ( no)
  - The serializers: ( no)
  - The runtime per-record code paths (performance sensitive): ( no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: ( no)
  - The S3 file system connector: ( no)

## Documentation

  - Does this pull request introduce a new feature? ( no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java [hunk 1]
```diff
@@ -702,14 +698,14 @@ private void closeTerminal() {
         }
     }
 
-    private LineReader createLineReader(Terminal terminal) {
+    private LineReader createLineReader(Terminal terminal, Completer completer) {
```
Comment (XComp) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java:701]:
```suggestion
    private LineReader createLineReader(Terminal terminal, @Nullable Completer completer) {
```

**Review by XComp — APPROVED** [review]
Thanks @snuyanzin . The change looks reasonable. May you apply the nitty comment I added below, split the commit up into two (because they are actually unrelated) and rebase the PR?

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java (hunk 1 above)
Comment (snuyanzin) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java]: During rebase I realize that there was introduced a similar method with `boolean` withing this PR [1] a second arg telling if completer should be added or not... So this change is not actual anymore after that
[1] https://github.com/apache/flink/pull/20182
link to the method https://github.com/apache/flink/pull/20182/files#diff-d5511cf0b3c81a60ae476f8a36c37c723fd703eb2c44c556473cb0066502481aR681

**Comment by XComp** [comment]
The [test failure](https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=42453&view=logs&j=4eda0b4a-bd0d-521a-0916-8285b9be9bb5&t=2ff6d5fa-53a6-53ac-bff7-fa524ea361a9&l=46486) is related to FLINK-29315. It's already reported.

**Comment by snuyanzin** [comment]
@flinkbot run azure

**Comment by XComp** [comment]
@flinkbot run azure

**Pull request merged**